WordPress/gutenberg
# Update `rel` and `title` labels for navigation and submenu links.

## What?
Updating the `rel` and `title` labels for navigation and submenu links.

Fixes https://github.com/WordPress/gutenberg/issues/50182

## Why?

The difference between the "Link label" and "Link title" fields was unclear with the previous labels. And most people wouldn't interpret that these fields would be for `attributes`.

These settings are already in the context of a link, so having the word "link" as part of these two labels seemed redundant. That's why we're proposing to change these from `Link rel`  and `Link title` to `Rel attribute` and `Title attribute`.

## Testing Instructions
1. Open the site editor.
2. Edit the navigation menu.
3. Add a menu link and check the new labels.
4. Add a submenu link and check the new labels.

## Screenshots or screencast 

https://user-images.githubusercontent.com/252415/235432824-8d4c02b9-6fe2-4c05-9b9f-586ecfebf5f4.mov
